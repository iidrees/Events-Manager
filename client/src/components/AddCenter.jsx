
import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';
import { Redirect } from 'react-router-dom'

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { addCenter, imageUpload } from '../actions/addCentersAction';
import { centerDetails } from '../actions/centerDetailsAction';


/**
 * 
 * @param {event} e {e}
 * @class AddCenter
 * @extends {React.Component}
 */
class AddCenter extends React.Component {

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  onSubmit = (e) => {
    e.preventDefault();
    let centerData = this.state;
    const { dispatch } = this.props;
    if (!centerData.imgFile) {
      return dispatch(addCenter(centerData))
    }
    return dispatch(imageUpload(centerData));
  }

/**
 * @param {e} e {event}
 * @returns {any} any
 * @memberof AddCenter
 */
onImageChange = (e) => {
    e.preventDefault();

    let reader = new FileReader();
    let imgFile = e.target.files[0];

    if (imgFile.type === 'image/jpeg' || imgFile.type === 'image/png') {
      reader.onloadend = () => {
        this.setState({
          imgFile: imgFile,
          imgUrl: reader.result
        });
      }
      reader.readAsDataURL(imgFile);
    } else {
      alert('Please upload an image with the .jpeg or .png file format')
    }
  
  }


/**
 * 
 * 
 * @returns {String} return the HTML markup to be rendered 
 * @memberof AddCenter
 */
render () {

    const { createCenter, user } =  this.props;
    console.log('add center component >>>>', createCenter, ' and User reducer', user)
    return (
      <div>
        <div>
          {
            (!user.authenticated) &&
            <Redirect to='/signin' push />
          }
        </div>
        <NavBarMain />
        <div className="container">
          <div className="row">
            <div className="container" id="add-center-header">
              <div className="row">
                <div className="col-sm-12">              
                  <h1 className="text-center head-1" id="admin-add-head">Add New Event Centers Below </h1>
                  <p className="head-para text-center">
                  
                  </p>
                  <hr /> 
                </div>
              </div>
            </div>
      
           {(createCenter.status === 'Success' ) && <div className="alert alert-success" role="alert">
												<button type="button" className="close" data-dismiss="alert" aria-label="Close">
													<span aria-hidden="true">&times;</span>
												</button>
                        <strong>{createCenter.message}.</strong></div>}
                        {(createCenter.status === 'Unsuccessful' ) && <div className="alert alert-danger" role="alert">
												<button type="button" className="close" data-dismiss="alert" aria-label="Close">
													<span aria-hidden="true">&times;</span>
												</button>
												<strong>{createCenter.message}</strong><span> </span>
												<strong>{createCenter.error}.</strong></div>}
            <div className="container">{/* <!-- Start container for Add Form --> */}
              <div className="row">
                <div className="col-md-6">
                  <form method="POST" className="form form-center" onSubmit={this.onSubmit} role="form" id="admin-form"  action="#">{/* <!-- ADD FORM --> */}
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Name of Center:</label>
                      <input className="form-control" onChange={this.onChange} name="name" type="text" placeholder="Enter Name of Center" id="example-text-input" />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Location:</label>
                      <input className="form-control" type="text" placeholder=" Enter Location" id="example-text-input" name="location" onChange={this.onChange}  />
                    </div>                
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Address:</label>
                      <input className="form-control" type="text" placeholder="Enter Address" id="example-text-input" name="address" onChange={this.onChange}/>
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Capacity:</label>
                      <input className="form-control" type="text" placeholder="Enter Capacity" id="example-text-input" name="capacity" onChange={this.onChange} />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Owner:</label>
                      <input className="form-control" type="text" placeholder="Enter Owner" id="example-text-input" name="owner" onChange={this.onChange} />
                    </div>
                    <div className="form-group">
                      <label htmlFor="event-details" className=" home-para"> Description:</label>
                      <textarea className="form-control" id="eventTextarea" rows="8" placeholder="brief details about the event" name="description" onChange={this.onChange} ></textarea>
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Upload an Image of event center below:</label>
                      <input type="file" className="form-control-file" onChange={this.onImageChange} id="input-file" name="images" aria-describedby="fileHelp" />
                    </div>  
                    <button type="submit" className="btn btn-primary btn-sm" id="save-event">Save and create Event Center<span><i className="fa fa-paper-plane" aria-hidden="true"></i></span></button>
                  </form>          
                </div>{/* <!-- End Container ADD EVENTS FORM --> */}
              </div>
            </div>
          </div>
        </div>
      </div>
    )

  }
}


const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    createCenter: state.addCenterReducer,
    user: state.userReducer
    
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(AddCenter);