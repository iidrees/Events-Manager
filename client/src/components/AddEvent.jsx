import React from 'react';
import { connect } from 'react-redux';
import { Redirect, withRouter } from 'react-router-dom';
import axios from 'axios';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';
import validator from 'validator';
import { Wave, Third } from 'react-preloading-component';

import NavBarMain from './NavBarMain.jsx';
import Footer from './Footer.jsx';
import { addEvent, imageUpload } from '../actions/addEvent';
import getCenters from '../actions/getCenters';
import { history } from '../routes';
import AddEventComponent from './EventsComponents/AddEventComponent.jsx';
import AddEventHeaderComponent from './EventsComponents/AddEventHeaderComponent.jsx';

/**
 * @param {event} event {events}
 * @class Addevents
 * @extends { React.Component }
 */
class AddEvent extends React.Component {
  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onSubmit = event => {
    event.preventDefault();
    let eventData = this.state;

    const { dispatch } = this.props;
    if (
      eventData.imgFile === null ||
      eventData.imgFile === undefined ||
      !eventData.imgFile
    ) {
      return dispatch(addEvent(eventData, eventData.center));
    }
    return dispatch(imageUpload(eventData, eventData.center));
  };

  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onImageChange = event => {
    event.preventDefault();

    let reader = new FileReader();
    let imgFile = event.target.files[0];

    try {
      if (imgFile.type === 'image/jpeg' || imgFile.type === 'image/png') {
        reader.onloadend = () => {
          this.setState({
            imgFile: imgFile,
            imgUrl: reader.result
          });
        };
        reader.readAsDataURL(imgFile);
      } else {
        alert('Please upload an image with the .jpeg or .png file format');
      }
    } catch (error) {
      alert('Please upload an image with the .jpeg or .png file format');
    }
  };

  /**
   * @returns {JSON} An array of centers
   * @memberof Addevents
   */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(getCenters());
  }

  /**
   * @returns {any} -
   * @memberof Addevents
   */
  render() {
    const { user, centers, event } = this.props;
    let userId, token, decoded;
    try {
      token = localStorage.getItem('x-access-token');

      userId = jwt.decode(token).id;

      if (jwt.decode(token).admin) {
        return <Redirect to="/getCenters" push />;
      }
    } catch (error) {
      decoded = null;
    }

    if (event.status === 'Unsuccessful') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error(`${event.message}`);
      event.status = '';
    }
    if (event.status === 'Success') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.success(`${event.message}`);
      event.status = '';
      return <Redirect to="/myevents" push />;
    }
    if (event.isLoading) {
      event.isLoading = false;
      return (
        <div style={{ paddingTop: '350px' }}>
          <Wave />
        </div>
      );
    }
    return (
      <div>
        <div className="container" id="add-events">
          <div className="row">
            <AddEventHeaderComponent />
            <div className="container">
              {/* <!-- Start container for Add Form --> */}
              <div className="row">
                <AddEventComponent
                  {...centers}
                  onChange={this.onChange}
                  onSubmit={this.onSubmit}
                  onImageChange={this.onImageChange}
                />
              </div>
            </div>
          </div>
          {/* <!-- End container for the Head section --> */}
        </div>
      </div>
    );
  }
}

/**
 * Maps dispatch to props
 * @param {any} dispatch -
 * @returns {void}
 */
const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};

/**
 * Maps state to props and the store is exposed to the component
 *
 * @param {any} state -
 * @returns {void}
 */
const mapStateToProps = state => {
  return {
    event: state.addEventReducer,
    user: state.userReducer,
    centers: state.centerReducer,
    token: state.userTokenReducer
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(AddEvent)
);
