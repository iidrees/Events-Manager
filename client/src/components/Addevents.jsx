import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { addEvent } from '../actions/addEventActions';
import  getCenters  from '../actions/getCentersAction';


/**
 * 
 * @class Addevents
 * @extends { React.Component }
 */
class Addevents extends React.Component {



  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  onSubmit = (e) => {
		e.preventDefault();
    let eventData = this.state;
   
		const { dispatch } = this.props;
		return dispatch(addEvent(eventData, eventData.center));
  }
  /* eslint-disable */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(getCenters());
  }

  /**
   * 
   * 
   * @returns {any} - 
   * @memberof Addevents
   */
  render() {
    
    const { centers } = this.props;
  return (
    <div>
    <NavBarMain />
    <div className="container">
    <div className="row">
      <div className="container" id="add-event-header">
        <div className="row">
          <div className="col-sm-12">              
            <h1 className="text-center head-1">Add your events </h1>
            <p className="head-para text-center">
              Are you in need of a location to host your event? 
              If yes, then why not host your events using one of 
              our numerous event centers by creating an event below?
            </p>
          </div>
        </div>
      </div>
    
      
      <div className="container">{/* <!-- Start container for Add Form --> */}
        <div className="row">
          <div className="col-sm-12">
            <form method="POST" className="form-horizontal" onSubmit={this.onSubmit} role="form"  action="#">{/* <!-- ADD FORM --> */}
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Name of Event:</label>
                <input className="form-control" onChange={this.onChange}  name="title" id="form-event1" type="text" placeholder=""  />
              </div>
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Date of event:</label>
                <input className="form-control" onChange={this.onChange}  name="date" type="text" id="form-event2"placeholder="YY/MM/DD"  />
              </div>              
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Time:</label>
                <input className="form-control" onChange={this.onChange}  name="time" type="text" id="form-event7"placeholder="hh:mm AM/PM"  />
              </div>
              <div className="form-group">
                <label htmlFor="event-centers" className=" home-para">Center:</label>
                <select className="form-control" id="event-center1" onChange={this.onChange}  name="center">
                  <option className="home-para">Choose an event center</option>
                  {(centers.map((center) => {
                    
                    return (<option key={center.id} value={center.id} className="home-para">{center.name}</option>)
                  }))}
                  
                </select>
              </div>
              <div className="form-group">
                <label htmlFor="event-details"  className=" home-para">Events description:</label>
                <textarea className="form-control" onChange={this.onChange} id="form-event4" rows="8" name="description" placeholder="brief details about the event"></textarea>
              </div>
              <div className="form-group">
                <label htmlFor="event-centers" className=" home-para">Event type:</label>                
                <input className="form-control" onChange={this.onChange}  name="type" type="text" id="event-center2" placeholder="What kind of event is it, 'public' or 'private'?"  />          
              </div>
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Upload an Image of your event below:</label>
                <input type="file" className="form-control-file" onChange={this.onChange}  aria-describedby="fileHelp" /> 
              </div>  
          
              <button type="submit" className="btn btn-primary btn-sm" id="save-event">Save and create event  <span><i className="fa fa-paper-plane" aria-hidden="true"></i></span></button>
            </form>          
          </div>{/* <!-- End Container ADD EVENTS FORM --> */}
        </div>
      </div>
  </div>{/* <!-- End container for the Head section --> */}
  </div>
  <Footer />
  </div>
  );
}
}


const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    addevents: state.eventReducer,
    status: state.userReducer,
    centers: state.centerReducer
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Addevents);