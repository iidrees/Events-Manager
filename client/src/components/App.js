import React from 'react';
import { Route, HashRouter, Switch, Link, BrowserRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import NavigationBar  from './NavigationBar.js';
import Main from './Main.jsx'
import Getevents from './Getevents.jsx'
import Footer from './footer.jsx';
import  './scss/style.scss';
import  './css/style.css';

/**
 * This is the App component that helps
 * abstract out the implementation of the sign-up component while using the react-router-dom
 * @class App
 * @extends {React.Component}
 */
class App extends React.Component {
  render() {
    const { userSignupRequest } = this.props
    return (
      <div className="container">
        <NavigationBar />
        <BrowserRouter>
          <Switch>            
            <Route exact path='/users' component={Main} />
            <Route exact path='/getEvents' component={Getevents} />
          </Switch>
        </BrowserRouter>
        <Footer />
      </div>
    );
  }
}
export default App;