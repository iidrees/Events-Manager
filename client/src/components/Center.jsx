import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect, withRouter } from 'react-router-dom';
import Pagination from 'rc-pagination';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';

import Footer from './Footer.jsx';
import getCenters from '../actions/getCentersAction';
import NavBarMain from './NavBarMain.jsx';
import { history } from '../routes';
import GetCentersComponent from './CentersComponents/GetCentersComponent.jsx';
import GetCentersHeaderComponent from './CentersComponents/GetCentersHeaderComponent.jsx';
/**
 * @class Center
 * @extends {React.Component}
 */
class Center extends React.Component {
  /**
   * Creates an instance of Center.
   * @param {any} props -
   * @memberof Center
   */
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1,
      itemsPerPage: 10,
      userName: '',
      isAdmin: false
    };
    this.onChange = this.onChange.bind(this);
  }

  /**
   * @returns {*} function
   * @memberof Center
   */
  componentDidMount() {
    const { dispatch, currentPage } = this.props;
    return dispatch(getCenters(currentPage));
  }
  /**
   *
   * @returns {any} -
   * @param {any} page -
   * @memberof Center
   */
  onChange = page => {
    this.setState(
      {
        currentPage: page
      },
      () => {
        const { dispatch } = this.props;
        return dispatch(getCenters(this.state.currentPage));
      }
    );
  };

  /**
   * @returns {*} any
   * @memberof Center
   */
  render() {
    const { centers, user, currentPage, itemsPerPage } = this.props;
    let userId, token, decoded;
    try {
      token = localStorage.getItem('x-access-token');
      userId = jwt.decode(token).id;
    } catch (error) {
      decoded = null;
    }
    if (centers.status === 'Unsuccessful') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error(
        'No Event Centers Available at the Moment, check back later'
      );
      centers.status = '';
    }
    if (centers.status === 'Success') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.success('These are the event centers available');
      centers.status = '';
    }
    return (
      <div>
        <div className="container">
          <div>
            <GetCentersHeaderComponent centers={centers} />
            <GetCentersComponent centers={centers} userId={userId} />
          </div>
        </div>
        <div className="col-sm-12 col-md-6 col-lg-4 paginator">
          {
            <Pagination
              onChange={this.onChange}
              current={this.state.currentPage}
              className="pagination"
              total={centers.count}
              locale={{ items_per_page: 'items' }}
            />
          }
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};
const mapStateToProps = state => {
  return {
    centers: state.centerReducer,
    user: state.userReducer
  };
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Center));
