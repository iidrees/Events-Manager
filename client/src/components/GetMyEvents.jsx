import React from 'react';
import { connect } from 'react-redux';
import { Redirect, Link, withRouter } from 'react-router-dom';
import Pagination from 'rc-pagination';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';
import { Wave } from 'react-preloading-component';
import _ from 'lodash';

import NavBarMain from './NavBarMain.jsx';
import Footer from './Footer.jsx';
import { getMyEvents } from '../actions/event';
import { history } from '../routes';

import EventComponent from './EventsComponents/EventComponent.jsx';
import MyEventHeader from './EventsComponents/MyEventHeader.jsx';

/**
 * Gets all user events
 * @class GetMyEvents
 * @extends {React.Component}
 */
export class GetMyEvents extends React.Component {
  /**
   * Creates an instance of Center.
   * @param {any} props -
   * @memberof GetMyEvents
   */
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1,
      itemsPerPage: 10
    };
    this.onChange = this.onChange.bind(this);
  }

  /**
   * A method that gets user events
   * @memberof GetMyEvents
   * @returns {void}
   */
  componentDidMount() {
    const { dispatch } = this.props;

    return dispatch(getMyEvents());
  }

  /**
   *
   * @returns {any} -
   * @param {any} page -
   * @memberof GetMyEvents
   */
  onChange = page => {
    this.setState(
      {
        currentPage: page
      },
      () => {
        const { dispatch } = this.props;
        return dispatch(getMyEvents(this.state.currentPage));
      }
    );
  };

  /**
   * @returns {JSX} -
   * @memberof GetMyEvents
   */
  render() {
    const { myEvents, status, isLoading } = this.props;
    let userId, token, decoded;
    try {
      token = localStorage.getItem('x-access-token');

      userId = jwt.decode(token).id;

      if (jwt.decode(token).admin) {
        return <Redirect to="/getCenters" push />;
      }
    } catch (error) {
      decoded = null;
    }

    if (myEvents.status === 'Unsuccessful' || !myEvents.events) {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error('You currently have no events');
      myEvents.status = '';
    }

    if (myEvents.status === 'Success') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.success(`${myEvents.message}`);
      myEvents.status = '';
    }

    return (
      <div>
        {_.isEmpty(myEvents.events) ? (
          //TODO: I still need to test this
          <h1
            style={{
              paddingTop: '60px',
              fontSize: '20px'
            }}
          >
            <div
              className="alert alert-success"
              role="alert"
              style={{ textAlign: 'center' }}
            >
              <strong style={{ textAlign: 'center' }}>You Have No Event</strong>
            </div>
          </h1>
        ) : (
          <div>
            <div className="container" id="myevent">
              <div className="row">
                <MyEventHeader myEvents={myEvents} />
                <EventComponent
                  onChange={this.onChange}
                  userId={userId}
                  events={myEvents}
                />
              </div>
            </div>
            <div className="col-sm-12 col-md-6 col-lg-4 paginator">
              {
                <Pagination
                  onChange={this.onChange}
                  current={this.state.currentPage}
                  total={myEvents.count}
                  locale={{ items_per_page: 'items' }}
                />
              }
            </div>
            <div id="card-body1" />
          </div>
        )}
        <Footer />
      </div>
    );
  }
}

/**
 *
 * Maps dispatch to props
 * @param {any} dispatch -
 * @returns {void}
 */
const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};

/**
 *
 * Maps state and the store to props
 * @param {any} state -
 * @returns {void}
 */
const mapStateToProps = state => {
  return {
    myEvents: state.myEventsReducer,
    status: state.userReducer,
    token: state.userTokenReducer
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(GetMyEvents)
);
