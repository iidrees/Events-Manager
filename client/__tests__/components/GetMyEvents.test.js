import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';

import sinon from 'sinon';

import { GetMyEvents } from '../../src/components/GetMyEvents';

describe('GetMyEvents component', () => {
  let props = {
    // status: '',
    myEvents: {
      status: 'Success',
      message: 'message',
      authenticated: true,
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJp' +
        'ZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluI' +
        'jp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2M' +
        'zU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI'
    },
    onChange: () => {},
    onSubmit: () => {},
    dispatch: () => {}
  };
  let wrapper = shallow(<GetMyEvents {...props} history={history} />);

  it('should render without throwing an error', () => {
    expect(shallow(<GetMyEvents {...props} />).length).toEqual(1);
  });

  it('should render the whole component when there are no events', () => {
    props = {
      // status: '',
      myEvents: {
        status: 'Unsuccessful',
        message: 'message',
        authenticated: true,
        token:
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnV' +
          'lLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXB' +
          'lckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJl' +
          'eHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI'
      },
      onChange: () => {},
      dispatch: () => {}
    };
    wrapper = shallow(<GetMyEvents {...props} />);
    expect(wrapper).toMatchSnapshot();
    const action = wrapper.instance();
    const onChange = jest.spyOn(wrapper.instance(), 'onChange');
    action.onChange();
    expect(onChange).toBeCalled();
    expect(wrapper.find('h1').length).toBe(1);
  });
  it('should render the whole component when there are no events', done => {
    props = {
      // status: '',
      myEvents: {
        status: 'Success',
        message: 'message',
        authenticated: true,
        events: [
          {
            title: 'The Amity party',
            time: '10:00 PM',
            date: '21-03-2019',
            type: 'public',
            center: 'The Multi-purpose',
            description:
              'This is going to be the weirdest parry on the month of March',
            imgUrl:
              'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
          },
          {
            id: 1,
            title: 'The Amity party',
            time: '10:00 PM',
            date: '21-03-2019',
            type: 'public',
            center: 'The Multi-purpose',
            description:
              'This is going to be the weirdest parry on the month of March',
            imgUrl:
              'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
          }
        ],
        token:
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0c' +
          'nVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNT' +
          'dXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkz' +
          'LCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI'
      },
      onChange: () => {},
      dispatch: () => {}
    };
    wrapper = shallow(<GetMyEvents {...props} />);
    const action = wrapper.instance();
    const onChange = jest.spyOn(wrapper.instance(), 'onChange');
    action.onChange();
    expect(onChange).toBeCalled();
    expect(wrapper.find('#myevent').length).toEqual(1);
    done();
  });
});
