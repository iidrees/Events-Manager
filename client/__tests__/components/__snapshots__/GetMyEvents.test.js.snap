// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetMyEvents component should render the whole component when there are no events 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetMyEvents
    dispatch={[Function]}
    myEvents={
      Object {
        "authenticated": true,
        "message": "message",
        "status": "",
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI",
      }
    }
    onChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
            Object {
              "fontSize": "20px",
              "paddingTop": "60px",
            }
          }
        >
          <div
            className="alert alert-success"
            role="alert"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <strong
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              You Have No Event
            </strong>
          </div>
        </h1>,
        <Unknown />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="alert alert-success"
            role="alert"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <strong
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              You Have No Event
            </strong>
          </div>,
          "style": Object {
            "fontSize": "20px",
            "paddingTop": "60px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              You Have No Event
            </strong>,
            "className": "alert alert-success",
            "role": "alert",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You Have No Event",
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "You Have No Event",
            "type": "strong",
          },
          "type": "div",
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
              Object {
                "fontSize": "20px",
                "paddingTop": "60px",
              }
            }
          >
            <div
              className="alert alert-success"
              role="alert"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <strong
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                You Have No Event
              </strong>
            </div>
          </h1>,
          <Unknown />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="alert alert-success"
              role="alert"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <strong
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                You Have No Event
              </strong>
            </div>,
            "style": Object {
              "fontSize": "20px",
              "paddingTop": "60px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                You Have No Event
              </strong>,
              "className": "alert alert-success",
              "role": "alert",
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You Have No Event",
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "You Have No Event",
              "type": "strong",
            },
            "type": "div",
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
